src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -157,14 +162,21 @@ const useCoreChatSystemHook = () => {
 
     if (isSystemMessage && isContainFollowingMessage) {
       const textOwnUser = 'You started following this user.\n Send them a message now.';
-      await ChannelList.updateChannelDescription(
-        localDb,
-        websocketData?.channel_id,
-        websocketData?.message?.textOwnMessage ?? textOwnUser,
-        websocketData
-      );
-      const chat = ChatSchema.fromWebsocketObject(websocketData);
-      await chat.save(localDb);
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveFollowingMessage,
+        id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+        priority: QueueJobPriority.HIGH,
+        task: async () => {
+          await ChannelList.updateChannelDescription(
+            localDb,
+            websocketData?.channel_id,
+            websocketData?.message?.textOwnMessage ?? textOwnUser,
+            websocketData
+          );
+          const chat = ChatSchema.fromWebsocketObject(websocketData);
+          await chat.save(localDb);
+        }
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The database operations are now being added to a priority queue instead of being executed immediately. This is a good practice for managing resources and preventing potential bottlenecks when there are many operations to perform. However, make sure that the priority queue is well-managed and that high-priority tasks do not starve lower-priority tasks.

```diff
- await ChannelList.updateChannelDescription(
-   localDb,
-   websocketData?.channel_id,
-   websocketData?.message?.textOwnMessage ?? textOwnUser,
-   websocketData
- );
- const chat = ChatSchema.fromWebsocketObject(websocketData);
- await chat.save(localDb);
+ queue.addPriorityJob({
+   operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveFollowingMessage,
+   id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+   priority: QueueJobPriority.HIGH,
+   task: async () => {
+     await ChannelList.updateChannelDescription(
+       localDb,
+       websocketData?.channel_id,
+       websocketData?.message?.textOwnMessage ?? textOwnUser,
+       websocketData
+     );
+     const chat = ChatSchema.fromWebsocketObject(websocketData);
+     await chat.save(localDb);
+   }
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->